plugin/rewrite/rewrite.go

```diff
@@ -44,7 +44,8 @@ func (rw Rewrite) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	for _, rule := range rw.Rules {
 		switch result := rule.Rewrite(ctx, state); result {
 		case RewriteDone:
-			if !validName(state.Req.Question[0].Name) {
+			_, validName := dns.IsDomainName(state.Req.Question[0].Name)
```

Comment:
the `if` can be pulled up, and just using `ok` is more go like:
~~~
if _, ok := dns.IsDomainName; !ok {

}
~~~
also the `x :=` seemed superfluous and why are we logging this *and* return the error, seem we can drop the `log.Errrof` here as well

Reply:
Regarding `x :=` - we we need to save the name before overwriting `Question[0]` with `originalQuestion`, the name is needed to construct the error message to be returned by `ServeDNS()`. I'm going to update the code